### PR TITLE
Core/Events: announce event system improvement.

### PR DESCRIPTION
**Pull request by Viste**
##### CHANGES PROPOSED:
-  Remove ugly hack with switch and cases. make selection through DB
###### ISSUES ADDRESSED:
Closes : part of #1187 and #1118
##### Target branch(es): Master

How to test:
if in DB in announce colum we have 0 dont announce the announce about event shouldnt start
if we have 1 announce should be even if in config he off
and if we have 2 value be gets from config

### CODE REVIEW

File: src/server/game/Events/GameEventMgr.cpp [hunk 1]
```diff
@@ -1189,14 +1190,9 @@ void GameEventMgr::UnApplyEvent(uint16 event_id)
...
-    switch (sWorld->getIntConfig(CONFIG_EVENT_ANNOUNCE))
-    {
-        case 0:                                             // disable
-            break;
-        case 1:                                             // announce events
-            sWorld->SendWorldText(LANG_EVENTMESSAGE, mGameEvent[event_id].description.c_str());
-            break;
-    }
+    uint8 announce = mGameEvent[event_id].announce;
+    if (announce == 1 || (announce == 2 && sWorld->getBoolConfig(CONFIG_EVENT_ANNOUNCE)))
```
Comment (Nefertumm) [inline src/server/game/Events/GameEventMgr.cpp]:
```suggestion
     if (announce == 1 || (announce == 2 && sWorld->getIntConfig(CONFIG_EVENT_ANNOUNCE)))
```

Comment (Viste) [reply src/server/game/Events/GameEventMgr.cpp]: :(

**Comment by BarbzYHOOL** [comment]
That looks very useful. Any example with an existing script though?

**Comment by Viste** [comment]
@BarbzYHOOL this is not related to game, only make struct of DB same as trinity and this is more clean when it do throug DB then as now by switch

File: data/sql/updates/pending_db_world/rev_1546538959760274673.sql [hunk 2]
```diff
@@ -0,0 +1,4 @@
+INSERT INTO version_db_world (`sql_rev`) VALUES ('1546538959760274673');
+
+ALTER TABLE `game_event`
+ADD COLUMN `announce` tinyint(3) unsigned NULL DEFAULT 2 COMMENT '0 dont announce, 1 announce, 2 value from config' AFTER `world_event`;
```
Comment (FrancescoBorzi) [inline data/sql/updates/pending_db_world/rev_1546538959760274673.sql]: what does the `NULL` after `unsiged` do?

Comment (Viste) [reply data/sql/updates/pending_db_world/rev_1546538959760274673.sql]: i miss not null >_<

**Comment by BarbzYHOOL** [comment]
> @BarbzYHOOL this is not related to game, only make struct of DB same as trinity and this is more clean when it do throug DB then as now by switch

I know... but an example of how it is used in the CODE would be kind of nice but if it's ported from TC then ok fine...

**Comment by Viste** [comment]
> > @BarbzYHOOL this is not related to game, only make struct of DB same as trinity and this is more clean when it do throug DB then as now by switch
> 
> I know... but an example of how it is used in the CODE would be kind of nice but if it's ported from TC then ok fine...

it used only as — 0 dont announce, 1 announce, 2 value from config, and only here. Its need for core can understand need send announce when event start or not.

**Comment by Viste** [comment]
How to test:
if in DB in announce colum we have 0 dont announce the announce about event shouldnt start 
if we have 1  announce should be even if in config he off
and if we have  2 value be gets from config

**Comment by wetbrownsauce** [comment]
Tested. 

- Set some game_events to occur every minute for easy testing.
- When announce = 2, events only announce in-game if config option is on.
- When announce = 1, events announce in-game even if config option is off.
- When announce = 0, events never announce in-game.

**Comment by BarbzYHOOL** [comment]
thank you guys

**Pull request merged**